src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx

```diff
@@ -0,0 +1,176 @@
+export const IdleOverview = () => {
+  const { idle: api } = useIdle()
+  const translate = useTranslate()
+  const toast = useToast()
+  const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const [opportunity, setOpportunity] = useState<IdleOpportunity | null>(null)
+  const [claimableTokens, setClaimableTokens] = useState<ClaimableToken[]>([])
+  const { chainId, contractAddress: vaultAddress, assetReference } = query
+
+  const assetNamespace = 'erc20'
+  const assetId = toAssetId({ chainId, assetNamespace, assetReference })
+  const vaultTokenId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference: vaultAddress,
+  })
+  const asset = useAppSelector(state => selectAssetById(state, vaultTokenId))
+  const underlyingToken = useAppSelector(state => selectAssetById(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+  // user info
+  const balance = useAppSelector(state =>
+    selectPortfolioCryptoBalanceByAssetId(state, { assetId: vaultTokenId }),
+  )
+
+  const cryptoAmountAvailable = bnOrZero(balance).div(`1e${asset.precision}`)
+  const fiatAmountAvailable = bnOrZero(cryptoAmountAvailable).times(marketData.price)
+
+  const selectedLocale = useAppSelector(selectSelectedLocale)
+  const descriptionQuery = useGetAssetDescriptionQuery({ assetId, selectedLocale })
+
+  const walletAddress = useWalletAddress('0x0000000000000000000000000000000000000000')
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        if (!(vaultAddress && api)) return
+        const [opportunity] = await Promise.all([
+          api.findByOpportunityId(
+            toAssetId({ chainId, assetNamespace, assetReference: vaultAddress }),
+          ),
+        ])
+        if (!opportunity) {
+          return toast({
+            position: 'top-right',
+            description: translate('common.somethingWentWrongBody'),
+            title: translate('common.somethingWentWrong'),
+            status: 'error',
+          })
+        }
+        setOpportunity(opportunity)
+
+        const claimableTokens = await opportunity.getClaimableTokens(walletAddress)
+        setClaimableTokens(claimableTokens)
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('IdleOverview error:', error)
+      }
+    })()
+  }, [api, vaultAddress, chainId, toast, translate, walletAddress])
+
+  const additionalParams: Record<string, any> = {}
+
+  useAppSelector(selectorState => {
+    if (claimableTokens && claimableTokens.length > 0) {
+      let rewardAssets: any[] = []
+      claimableTokens.forEach(token => {
+        const rewardAsset = selectAssetById(selectorState, token.assetId)
+        if (rewardAsset) {
+          rewardAssets.push({
+            ...rewardAsset,
+            cryptoBalance: bnOrZero(token.amount).div(`1e+${asset.precision}`).toPrecision(),
+          })
+        }
+      })
+
+      additionalParams.rewardAssets = rewardAssets
+    }
+  })
```

Comment:
Same thing as before:
- no explicit any
- never mutate things in the context of a JSX component, that's what `setState()` is for. This will either cause excessive rerenders or not behave the way you think it does, since react uses triple equal comparison, the reference will still be the same